src/ResolveUserDecorator.php

```diff
@@ -96,4 +88,14 @@ public function getEventSubscriberClass()
     {
         return get_class($this->wrapped);
     }
+    
+    /**
+    * Get current Auth manager.
+    *
+    * @return \Illuminate\Contracts\Auth\Factory
+    */
+    protected function getAuth()
+    {
+    	return app('auth');
```

Comment:
Perhaps better to return the guard here straight away.
`->user()` doesn't exist on `\Illuminate\Contracts\Auth\Factory`. (Probably through a magic call, but rather have it explicit)

Reply:
Fixed.  Yes, the default auth manager passes __call() straight to the guard.  I agree that explicit is better.